reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+In most usecases, the migration will look like replacing imports inside of an experiment script.
+
+For example:
+
+```python
+from gymnasium.wrappers import TimeLimit
+
+import ray
+from ray import air
+from ray import tune
+from ray.rllib.examples.env.cartpole_mass import CartPoleMassEnv
+from ray.tune.registry import register_env
+
+# from ray.rllib.algorithms.maml import MAML, MAMLConfig  # BEFORE
+from rllib_maml.maml import MAML, MAMLConfig  # AFTER
+
+if __name__ == "__main__":
+    ray.init()
+    register_env(
+        "cartpole",
+        lambda env_cfg: TimeLimit(CartPoleMassEnv(), max_episode_steps=200),
+    )
+
+    config = MAMLConfig()
+
+    tuner = tune.Tuner(
+        MAML,
+        param_space=config.to_dict(),
+        run_config=air.RunConfig(stop={"training_iteration": 100}),
+    )
+    results = tuner.fit()
+
+```
+
+
+This proposed algorithms to remove are:
+- A3C
+- AphaStar
+- AlphaZero
+- ApexDQN
+- ApexDDPG
+- DDPG
+- ARS
+- Bandits 
+- CRR 
+- DDPG
+- DDPPO
+- Dreamer
+- DT
+- ES
+- Leelachess
+- MADDPG
+- MAML
+- PG
+- QMix
+- Random
+- SimpleQ
+- SlateQ
+- TD3
+- R2D2
+
+
+### Shepherd of the Proposal (should be a senior committer)
+
+@sven1977
+
+## Design and Architecture
+
+- We introduce the new  repo for deprecated RLlib algorithms and community contributed algorithms.
+
+### `rayproject/rllib-contrib` File Structure
+
+```
+README.md
+.github/
+    workflow/
+        algorithm_foo.yml <---------- This will launch the unit test for algorithm foo with gh actions
+algorithm_foo/
+    src/
+        algorithm_foo.py
+    tests/
+    examples/
+    requirements.txt
+    pyproject.toml  <---------- Used for installation
+
+latest_results/
+    algorithm_foo/
+        results.json
+        results.csv
+        results.pkl
+        results.tensorboard
+        results.md  <---------- This will contain some pictures of the performance as a sanity check
+```
+
+### Installation
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+
+## Contribution Guidlines
+
+
+The RLlib team commits to the following level of support for the algorithms in this repo:
+
+| Platform | Purpose | Estimated Response Time | Support Level |
+| --- | --- | --- | --- |
+| [Discuss Forum](https://discuss.ray.io) | For discussions about development and questions about usage. | < 1 week | Community |
```

Comment:
As discussed offline, I propose that we have a new section on the forums.
We can easily move questions from RLlib to there.
My hope is that we can create more separation between the cleaner, more performant, more well-tested part of RLlib and the contrib part for our community.

Reply:
I would increase the response time as the investigating questions and errors needs often longer due to different requirements etc.